app/controllers/api/lists_controller.rb

```diff
@@ -15,11 +15,22 @@ def destroy
     user = User.find(params[:user_id])
     list = user.lists.find(params[:id])
     list.destroy
-    render json: { :message => :"List #{params[:id]} has been deleted" }, status: :ok
+    render json: { message: :"List #{params[:id]} has been deleted" }, status: :ok
   rescue ActiveRecord::RecordNotFound
     render json: { error: 'List not found' }, status: :not_found
   end
 
+  def update
+    list = List.find(params[:id])
+    if list.public? && list.update(list_params)
+      render json: list
+    elsif list.public == false
+      render json: { error: :"This list is private and cannot be edited."}, status: 400
+    else
+      render json: { errors: list.errors.full_messages }, status: :unprocessable_entity
```

Comment:
That is a great question. I was wondering that myself since I checked for the available messages and got an empty array (or hash?). I was reading about error handling in the ruby guides and it sounded like I need to actually add specific messages. Is that correct?

> On Jan 8, 2016, at 8:43 PM, David Roberts <notifications@github.com> wrote:
> 
> In app/controllers/api/lists_controller.rb:
> 
> >    rescue ActiveRecord::RecordNotFound
> >      render json: { error: 'List not found' }, status: :not_found
> >    end
> >  
> > +  def update
> > +    list = List.find(params[:id])
> > +    if list.public? && list.update(list_params)
> > +      render json: list
> > +    elsif list.public == false
> > +      render json: { error: :"This list is private and cannot be edited."}, status: 400
> > +    else
> > +      render json: { errors: list.errors.full_messages }, status: :unprocessable_entity
> What conditions would cause this line to run?
> 
> —
> Reply to this email directly or view it on GitHub.
>